src/EurekaClient.js

```diff
@@ -426,10 +430,22 @@ export default class Eureka extends EventEmitter {
   eurekaRequest(opts, callback, retryAttempt = 0) {
     this.clusterResolver.resolveEurekaUrl((err, eurekaUrl) => {
       if (err) return callback(err);
-      const requestOpts = merge({
+      let requestOpts = merge({
         baseUrl: eurekaUrl,
         gzip: true,
       }, opts);
+
+      if (this.requestMiddleware) {
+        try {
+          requestOpts = this.requestMiddleware(requestOpts);
```

Comment:
I was wondering... would it be better to convert this into something that uses continuations rather than returning the request object? The main use-case I wanted to get out of this change was being able to use this middleware to accommodate acquiring an OAuth2 access token. Since that is an asynchronous operation it would not be possible to do that here unless the format of the middleware were something more like:

```js
function (request, done) {
    // Can go make a request to OAuth server to retrieve an access token here
    ...
    // Finally once token is acquired
    request.headers['Authorization'] = `Bearer ${token}`;
    done(request);
}
```